### ISSUE TITLE
Some windows go black when moving them

### ISSUE THREAD

**Issue by Naboum** [digitalcreations/MaxTo#135]
Hi, I noticed that when using MaxTo, some applications are showing a black or white screen when I move them. I uploaded a video which is easier to explain to you what's happening : 

[Here](http://webmshare.com/play/Mnodr)

I encountered this issue with only 2 applications for now : Nvidia Geforce Experience and Uplay (which is showing a white window instead).

As you can see, as soon as I start moving around the window, it goes black, but I can still click on the close button in top right corner of the window, which means that the application doesn't crash or stop responding, it's just a visual issue.

When I close MaxTo, I don't have this problem.

I'm using Windows 10 64bit

**Comment by vegardlarsen** [comment]
@Naboum Let me first apologize for the slow response time. I have been on vacation for the last two weeks, so I haven't been able to respond.

Does this happen with other apps as well?

You could try to figure out what the name of the executable is (open up Task Manager, see the Name column in the Details tab). When you have the executable name, open up Notepad as Administrator, and edit `C:\Program Files (x86)\MaxTo\shims.xml`. 

I think the executable name is `gfexperience.exe` (please confirm if I am right), so remove the file ending and under `<no_subclass>` add `<process title="Nvidia GeForce Experience">gfexperience</process>`. Save the file and restart MaxTo.

if this works, please let me know the application names you do this with, so I can make this work for everyone in a future release.

**Comment by Naboum** [comment]
I added these lines 
```
    <process title="NVIDIA GeForce Experience">GFExperience</process>
	<process title="Uplay launcher">Uplay</process>
```
but it doesn't seem to work.

For now these are the only two apps I had problem with.
And in the task manager, the G and F are in capital letters, I don't know if it's important to say it.

**Comment by vegardlarsen** [comment]
Could you post your entire shims.xml here? The process name is case insensitive.

**Comment by Naboum** [comment]
Sure
[shims.zip](https://github.com/digitalcreations/MaxTo/files/256050/shims.zip)

**Comment by vegardlarsen** [comment]
Try adding them under `<ignore>` instead, and see if that helps?

**Comment by Naboum** [comment]
Ok so now there are a few advancements.

When I launch Uplay.exe, and move the window, it doesn't bug. If I minimize the window in the task bar, restore it then move around the window again, still no white/black window issue.

Actually, it happens only when I start the exe, minimize it in the system tray, restore it and move the window.
So the step that seems to cause the visual issue here is when the exe is minimized in the system tray.

**Comment by AssaultCommand** [comment]
Disable transparency when moving windows in the settings, I had the exact same issue you're having until I did this. Something with the transparency not wanting to play nice with custom windows.

**Comment by Naboum** [comment]
> Disable transparency when moving windows in the settings, I had the exact same issue you're having until I did this. Something with the transparency not wanting to play nice with custom windows.

That was exactly the problem, thank you :)

**Comment by vegardlarsen** [comment]
Good catch! I may have to add a compatibility shim that disables transparency effects on specific windows. :)

**Comment by cppcooper** [comment]
Naboum, good catch. I've been wondering what causes Uplay to do that, Cmder had a problem with transparency as well; but I had never put it together what they had in common when it happened. Many thanks to you for adding it to the internet as well. 👍

I agree a blacklist is a good idea. I don't mind having the feature but I'd disable it over keeping this Uplay issue. Maybe somebody should report a bug to them.

**Comment by okand** [comment]
Hi. Sorry to bump an old thread but I've started having this issue recently. More specifically it is my password manager, 1Password, whose window goes completely white if I move or resize it while the transparency option in MaxTo is enabled.

For a long while they worked just fine together but in the last month or two 1Password has started doing this. I can still aim in the top right corner to close the window and then reopen the app in the position and at the size I left it in but that is an annoying extra step to have to take every time I want to move 1Password to somewhere else on the screen.

I'm using version 2.0.1 with the updater set to beta.

**Comment by okand** [comment]
I had to change the second to last line to be capitalised `1Password` instead of just compying your example exactly but that works. The 1Password window no longer tries to go transparent at all and therefore also doesn't go white.

**Comment by vegardlarsen** [comment]
Thank you for your feedback. I've put 1Password onto our [official compatibility shims](https://maxto.net/no/shim/1password), so everyone should get an update shortly. You can update manually now from Settings, Compatibility. Thank you for helping me figure this out. :)

**Comment by Jiblet** [comment]
Hi, I found this thread while researching a problem with MaxTo playing nicely with UPlay - it was mentioned at the top of this thread but I don't think a solution was posted... until now!

- Open %LocalAppData%\MaxTo\shims.json in Notepad (or any other text editor)

- Add the following entry to the bottom of the file:

```json
    "id": "Uplay",
    "descriptions": {
      "en": {
        "title": "Uplay",
	    "subtitle": "Uplay client going white when dragging - issue with transparency.",
	    "description": null
      }
    },
    "iconUrl": null,
    "url": null,
    "canDisable": true,
    "defaultEnabled": true,
    "ignore": false,
    "disableSubclassing": false,
    "disableSnapping": false,
    "disableTransparency": true,
    "useMoveWindow": false,
    "emulatesMaximize": false,
    "windowClasses": ["uplay_main"],
    "processes": ["UplayWebCore.exe"]
  }
```

Restart MaxTo and restart UPlay if its done its trick of going white and all should be well with the world!

**Comment by vegardlarsen** [comment]
@Jiblet Thanks for that. I have included this in the default shims, so everyone else will get this automatically. :)
